pkg/datasources/streamlits_acceptance_test.go

```diff
@@ -0,0 +1,189 @@
+package datasources_test
+
+import (
+	"context"
+	"fmt"
+	"maps"
+	"regexp"
+	"testing"
+
+	acc "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/snowflakeroles"
+	"github.com/stretchr/testify/require"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/helpers/random"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider/resources"
+	"github.com/hashicorp/terraform-plugin-testing/config"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/tfversion"
+)
+
+func TestAcc_Streamlits(t *testing.T) {
+	databaseId := acc.TestClient().Ids.DatabaseId()
+	schemaId := acc.TestClient().Ids.SchemaId()
+	id := acc.TestClient().Ids.RandomSchemaObjectIdentifierInSchema(schemaId)
+	ctx := context.Background()
+	acc.TestAccPreCheck(t)
```

Comment:
this should be in the resource.Test (check other tests);

Reply:
This needs to be called earlier because there are resources being created in the provided schema.

Reply:
yeah, this is messy, but we know that it needs a better setup (like with integration tests); let's leave it like this for now